x/subscription/keeper/subscription.go

```diff
@@ -230,80 +236,86 @@ func (k Keeper) CreateSubscription(
 	return err
 }
 
+func (k Keeper) createNewSubscription(ctx sdk.Context, plan *planstypes.Plan, creator, consumer string,
+	block uint64, autoRenewalFlag bool,
+) (types.Subscription, error) {
+	sub := types.Subscription{
+		Creator:       creator,
+		Consumer:      consumer,
+		Block:         block,
+		PlanIndex:     plan.Index,
+		PlanBlock:     plan.Block,
+		DurationTotal: 0,
+		AutoRenewal:   autoRenewalFlag,
+	}
+
+	sub.MonthCuTotal = plan.PlanPolicy.GetTotalCuLimit()
+	sub.MonthCuLeft = plan.PlanPolicy.GetTotalCuLimit()
+	sub.Cluster = types.GetClusterKey(sub)
+
+	// new subscription needs a default project
+	err := k.projectsKeeper.CreateAdminProject(ctx, consumer, *plan)
+	if err != nil {
+		return types.Subscription{}, utils.LavaFormatWarning("failed to create default project", err)
+	}
+
+	return sub, nil
+}
+
+func (k Keeper) upgradeSubscriptionPlan(ctx sdk.Context, duration uint64, sub *types.Subscription, toPlanIndex string, toPlanBlock uint64) error {
+	date := ctx.BlockTime()
+	block := uint64(ctx.BlockHeight())
+
+	k.addCuTrackerTimerForSubscription(ctx, block, sub)
+
+	if sub.DurationLeft == 0 {
+		// Subscription was already expired. Can't upgrade.
+		k.handleZeroDurationLeftForSubscription(ctx, block, date, sub)
+		return utils.LavaFormatError("Trying to upgrade subscription, but subscription was already expired", nil,
+			utils.LogAttr("consumer", sub.Consumer))
+	}
+
+	// The "old" subscription's duration is now expired
+	// If called from CreateSubscription, the duration will reset to the duration bought
+	sub.DurationLeft = 0
```

Comment:
reset the DurationTotal

Reply:
Don't we want to include the pre-upgrade `DurationTotal` in the current one?

Reply:
nope, this is for the cluster, counts the month with the current plan index

Reply:
Got it. Fixed in cbe5433f810e664427fc95573e5c8b917f7b3f58